run_test.py

```diff
@@ -27,12 +27,17 @@ def main():
     for i in range(len(outputs)):
         if outputs[i] != excepted_outputs[i]:
             error += "Found some error...\n"
+    
+    #valgrind futása
+    for i in commands:
+        os.system("valgrind " + i)
+        if (os.system("echo $?") != 0):
+            error += "Memory leak...\n"
```

Comment:
A valgrindet olyan binárissal kell futtatni, ami nem ASannal lett lefordítva. És alapból ugyanaz az exit code-ja, ami a vizsgált programnak. Akkor jelzi nemnulla exit code-dal a memóriahibát, ha ez be van állítva az `--error-exitcode=<n>` kapcsolóval.